scripts/gen_ta_src.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/python3
+'''
+Script gen_ta_src.py used for generating source files which creates lookup table
+of loadable ta's. This table used by tee supplicant to load ta to OP-TEE
+Options:
+  -u [UUIDS ...], --uuids [UUIDS ...]
+                        List of enabled layers
+  -f FILE, --file FILE  Config file with enabled layers
+  -o OUT, --out OUT     Output file name (without extension)
+  -s SOURCE, --source SOURCE
+                        Directory with prebuilt ta
+  -t TABLE, --table TABLE
+                        table variable name
+List ot ta can be passed to script follows ways:
+  - with option -u can be pointed in command line
+  - with option -f list of uuids can be enumerated in text file
+  - with option -s script reads all files with extension .ta and used
+                   it as list of enabled uuids
+
+Script generates two files with extensions .c and .S and name pointed with -o option
+To use lookup table in application genarated files just should be build with application.
+'''
+import os
+import argparse
+
+def read_uuids_file(enabled_file):
+    uuids = []
+    for line in enabled_file:
+        line = line.strip()
+        if len(line) > 0 and line[0] != "#":
+            if line not in uuids:
+                uuids.append(get_components(line))
+    return uuids
+
+def get_components(uuid):
+    LENGTHS = [8, 4, 4, 4, 12]
+    items = uuid.split('-')
+    if len(items) != 5:
+        return []
+    components = []
+    i = 0;
+    while i<5:
+        item = items[i]
+        format = f'%0{LENGTHS[i]}x'
+        components.append(format % int(item, 16))
+        i += 1
+    return items;
+
+def compare_uuids(uuid1, uuid2):
```

Comment:
Again. Just `import uuid` and use the provided API

Reply:
updated